src/status_im/transport/message/v1/contact.cljs

```diff
@@ -0,0 +1,131 @@
+      (when (protocol/is-new? message-id)
+        (handlers/merge-fx cofx
+                           (contacts/receive-contact-update chat-id
+                                                            signature
+                                                            this))))))
+
+(handlers/register-handler-fx
+  ::send-new-sym-key
+  (fn [{:keys [db random-id] :as cofx} [_ {:keys [chat-id topic message sym-key sym-key-id]}]]
+    (let [{:keys [web3 current-public-key]} db]
+      (handlers/merge-fx cofx
+                         {:db (assoc-in db [:transport/chats chat-id :sym-key-id] sym-key-id)
+                          :shh/add-filter {:web3       web3
+                                           :sym-key-id sym-key-id
+                                           :topic      topic
+                                           :chat-id    chat-id}
+                          :data-store.transport/save {:chat-id chat-id
+                                                      :chat    (-> (get-in db [:transport/chats chat-id])
+                                                                   (assoc :sym-key-id sym-key-id)
+                                                                   ;;TODO (yenda) remove once go implements persistence
+                                                                   (assoc :sym-key sym-key))}}
+                         (message/send (NewContactKey. sym-key topic message)
+                                       chat-id)))))
+
+(handlers/register-handler-fx
+  ::add-new-sym-key
+  (fn [{:keys [db] :as cofx} [_ {:keys [sym-key-id sym-key chat-id topic message]}]]
+    (let [{:keys [web3 current-public-key]} db]
+      (handlers/merge-fx cofx
+                         {:db (assoc-in db [:transport/chats chat-id :sym-key-id] sym-key-id)
+                          :shh/add-filter {:web3       web3
+                                           :sym-key-id sym-key-id
+                                           :topic      topic
+                                           :chat-id    chat-id}
+                          :data-store.transport/save {:chat-id chat-id
+                                                      :chat    (-> (get-in db [:transport/chats chat-id])
+                                                                   (assoc :sym-key-id sym-key-id)
+                                                                   ;;TODO (yenda) remove once go implements persistence
+                                                                   (assoc :sym-key sym-key))}}
+                         (message/receive message chat-id chat-id)))))
+
+(handlers/register-handler-fx
+  :send-test-message
+  (fn [cofx [this timer chat-id n]]
+    (if (zero? n)
+      (println  "Time: " (str (- (inst-ms (js/Date.)) @timer)))
+      (handlers/merge-fx cofx
+                         {:dispatch [this timer chat-id (dec n)]}
+                         (message/send (protocol/map->Message {:content      (str n)
```

Comment:
` (message/send (protocol/map->Message` strikes me as weird somehow. I would expect it to be either:

`message/send m` or `send-foo-type-of-message m` (which deals with turning it into right format)